lib/android-helpers.js

```diff
@@ -76,23 +77,41 @@ helpers.ensureDeviceLocale = async function (adb, language, locale) {
   }
 };
 
-helpers.getActiveDevice = async function (adb, udid) {
+helpers.getDeviceInfoFromCaps = async function (opts = {}) {
+  // we can create a standalone ADB instance here, so there is no dependency
+  // on instantiating on earlier (at this point, we have no udid)
+  // we can only use this ADB object for commands that would not be confused
+  // if multiple devices are connected
+  let adb = await ADB.createADB({javaVersion: opts.javaVersion});
+  let udid = opts.udid;
+  let emPort = null;
+
   logger.info("Retrieving device list");
   let devices = await adb.getDevicesWithRetry();
-  let deviceId = null, emPort = null;
   if (udid) {
     if (!_.contains(_.pluck(devices, 'udid'), udid)) {
       logger.errorAndThrow(`Device ${udid} was not in the list ` +
                            `of connected devices`);
     }
-    deviceId = udid;
-    emPort = adb.getPortFromEmulatorString(deviceId);
+    emPort = adb.getPortFromEmulatorString(udid);
   } else {
-    deviceId = devices[0].udid;
-    emPort = adb.getPortFromEmulatorString(deviceId);
+    udid = devices[0].udid;
+    emPort = adb.getPortFromEmulatorString(udid);
+  }
+  logger.info(`Using device: ${udid}`);
+  return {udid, emPort};
+};
+
+// returns a new adb instance with deviceId set
+helpers.createADB = async function (javaVersion, udid, emPort) {
+  let adb = await ADB.createADB({ javaVersion });
```

Comment:
oh gothca. Was wondering if we treated objects differently when we used the "key is same as value" shorthand.